File: app/src/main/java/com/compuspulse/data/remote/model/Banner.java
package com.compuspulse.data.remote.model;

import com.google.gson.annotations.SerializedName;

public class Banner {
    private int id;
    private String title;
    @SerializedName("imagePath")
    private String imageUrl;
    @SerializedName("url")
    private String targetUrl;

    public Banner() {
    }

    public Banner(int id, String title, String imageUrl, String targetUrl) {
        this.id = id;
        this.title = title;
        this.imageUrl = imageUrl;
        this.targetUrl = targetUrl;
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title == null ? "" : title;
    }

    public String getImageUrl() {
        return imageUrl == null ? "" : imageUrl;
    }

    public String getTargetUrl() {
        return targetUrl == null ? "" : targetUrl;
    }
}

File: app/src/main/java/com/compuspulse/data/remote/model/Banner.kt
package com.compuspulse.data.remote.model

import com.google.gson.annotations.SerializedName

class Banner() {
    private var id: Int = 0
    private var title: String? = null

    @SerializedName("imagePath")
    private var imageUrl: String? = null

    @SerializedName("url")
    private var targetUrl: String? = null

    constructor(id: Int, title: String?, imageUrl: String?, targetUrl: String?) : this() {
        this.id = id
        this.title = title
        this.imageUrl = imageUrl
        this.targetUrl = targetUrl
    }

    fun getId(): Int = id

    fun getTitle(): String = title ?: ""

    fun getImageUrl(): String = imageUrl ?: ""

    fun getTargetUrl(): String = targetUrl ?: ""
}
